File: examples/android-host-shell/feature-bookmarks/src/main/java/com/doogat/hostshell/bookmarks/BookmarksModule.kt
package com.doogat.hostshell.bookmarks

import com.doogat.hostshell.ZDBModule
import uniffi.zdb_core.ZettelDriver

object BookmarksModule : ZDBModule {
    override val tables = listOf("category", "bookmark")

    override fun bootstrap(driver: ZettelDriver) {
        driver.executeSql("CREATE TABLE IF NOT EXISTS category (name TEXT NOT NULL)")
        driver.executeSql("""
            CREATE TABLE IF NOT EXISTS bookmark (
                title TEXT NOT NULL,
                url TEXT NOT NULL,
                category TEXT REFERENCES category(id)
            )
        """.trimIndent())
    }
}

File: examples/ios-host-shell/HostShell/Sources/Modules/Bookmarks/BookmarksModule.swift
import Foundation

struct BookmarksModule: ZDBModule {
    static let tables = ["category", "bookmark"]

    static func bootstrap(_ driver: ZettelDriver) throws {
        _ = try driver.executeSql(sql: "CREATE TABLE IF NOT EXISTS category (name TEXT NOT NULL)")
        _ = try driver.executeSql(sql: """
            CREATE TABLE IF NOT EXISTS bookmark (
                title TEXT NOT NULL,
                url TEXT NOT NULL,
                category TEXT REFERENCES category(id)
            )
        """)
    }
}
